public_html/wp-content/themes/pattern-directory/src/store/resolvers.js

```diff
@@ -16,16 +17,35 @@ import {
 	loadPatternFlagReasons,
 	loadPatterns,
 } from './actions';
-import { getQueryString } from './utils';
+import { PER_PAGE, getQueryString } from './utils';
+
+async function parseResponse( response ) {
+	try {
+		return {
+			total: response.headers?.get( 'X-WP-Total' ),
+			totalPages: response.headers?.get( 'X-WP-TotalPages' ),
+			results: await response.json(),
+		};
+	} catch ( error ) {
+		return {};
+	}
+}
 
 export function* getPatternsByQuery( query ) {
 	const queryString = getQueryString( query );
 	try {
 		yield fetchPatterns( queryString );
-		const results = yield apiFetch( {
-			path: addQueryArgs( '/wp/v2/wporg-pattern', query ),
+		const response = yield apiFetch( {
+			path: addQueryArgs( '/wp/v2/wporg-pattern', { ...query, per_page: PER_PAGE } ),
+			parse: false,
+		} );
+		const { total, totalPages, results } = yield __unstableAwaitPromise( parseResponse( response ) );
+		yield loadPatterns( queryString, {
+			page: query.page || 1,
+			patterns: results,
+			total: Number( total ),
```

Comment:
This is going to `NaN` if something goes wrong with parsing (`Number(undefined)`), I expect we would want to return `undefined` or `0` in that case. Maybe consider moving `Number()` into the `parseResponse` function.

Reply:
Done in https://github.com/WordPress/pattern-directory/pull/110/commits/4dbef7c0c97e8e9e8506ad2be4b2036f1ef690ce - Moved the `Number` conversion to `parseRequest`, with a fallback to 0.